src/cow-react/modules/swap/state/EthFlow/hooks.ts

```diff
@@ -0,0 +1,7 @@
+import { useDetectNativeToken } from 'state/swap/hooks'
+
+export function useIsEthFlow() {
+  const { isNativeIn, isWrapOrUnwrap, native } = useDetectNativeToken()
+
+  return { isEthFlow: isNativeIn && !isWrapOrUnwrap, nativeCurrency: native }
```

Comment:
Better to get rid of `nativeCurrency` from the return value, looks like `Single responsibility` breach